hydromt/components/table.py

```diff
@@ -0,0 +1,133 @@
+"""Table component."""
+
+import glob
+import os
+from os.path import basename, dirname, join
+from typing import (
+    TYPE_CHECKING,
+    Dict,
+    Optional,
+    Union,
+)
+
+import pandas as pd
+
+from hydromt.components.base import ModelComponent
+from hydromt.hydromt_step import hydromt_step
+from hydromt.utils.constants import DEFAULT_TABLE_FILENAME
```

Comment:
I disagree, I think that making it a class variable actual makes it less clear. Because you cannot give class variables as default arguments, so then you get more hidden functionality. Plugins can still override this by supplying a different default argument of their own. I'm fine with defining the constant in this file if you find that more clear, but I don't think making it a class variable makes sense

Reply:
Then let's leave it outside of the ModelComponent class. But maybe then define the variable directly in this file instead of in hydromt.utils.constant.
It would be great to have this consistent for all ModelComponents which is currently not the case. May I suggest the following:
```suggestion
DEFAULT_FILENAME = "tables/{name}.csv"
```
@Tjalling-dejong and @deltamarnix also need your opinion on this for your respective PRs.

Reply:
As marnix is still on holiday, I will be resolving this for now, but I've made a ticket for it (#864 ) and we will pick that up when he comes back